wrappers/IWearLogger/src/IWearLogger.cpp

```diff
@@ -0,0 +1,890 @@
+    wearable::VectorOfSensorPtr<const wearable::sensor::IForce3DSensor> force3DSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IForceTorque6DSensor> forceTorque6DSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IFreeBodyAccelerationSensor>
+        freeBodyAccelerationSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IGyroscope> gyroscopes;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IMagnetometer> magnetometers;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IOrientationSensor> orientationSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IPoseSensor> poseSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IPositionSensor> positionSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::ITemperatureSensor> temperatureSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::ITorque3DSensor> torque3DSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IVirtualLinkKinSensor>
+        virtualLinkKinSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IVirtualJointKinSensor>
+        virtualJointKinSensors;
+    wearable::VectorOfSensorPtr<const wearable::sensor::IVirtualSphericalJointKinSensor>
+        virtualSphericalJointKinSensors;
+
+    std::unordered_map<std::string, std::string> wearable2MatlabNameLookup;
+};
+
+IWearLogger::IWearLogger()
+    : yarp::os::PeriodicThread(DefaultPeriod)
+    , pImpl{new impl()}
+{}
+
+IWearLogger::~IWearLogger()
+{
+    detachAll();
+    close();
+}
+
+// ========================
+// PeriodicThread interface
+// ========================
+
+void IWearLogger::run()
+{
+    if (!pImpl->iWear) {
+        yError() << logPrefix << "The IWear pointer is null in the driver loop.";
+        askToStop();
+        return;
+    }
+
+    if (!pImpl->iPreciselyTimed) {
+        yError() << logPrefix << "The IPreciselyTimed pointer is null in the driver loop.";
+        askToStop();
+        return;
+    }
+
+    while (pImpl->iWear->getStatus() == WearStatus::Calibrating
+           || pImpl->iWear->getStatus() == WearStatus::WaitingForFirstRead) {
+        if (pImpl->waitingFirstReadCounter++ % 1000 == 0) {
+            pImpl->waitingFirstReadCounter = 1;
+            yInfo() << logPrefix << "IWear interface waiting for first data. Waiting...";
+        }
+        return;
+    }
+
+    if (pImpl->iWear->getStatus() == WearStatus::Error
+            || pImpl->iWear->getStatus() == WearStatus::Unknown) {
+        yError() << logPrefix << "The status of the IWear interface is not Ok ("
+                 << static_cast<int>(pImpl->iWear->getStatus()) << ")";
+        askToStop();
+        return;
+    }
+
+    // case status is TIMEOUT or DATA_OVERFLOW
+    if (pImpl->iWear->getStatus() != WearStatus::Ok) {
+        yWarning() << logPrefix << "The status of the IWear interface is not Ok ("
+                 << static_cast<int>(pImpl->iWear->getStatus()) << ")";
+    }
```

Comment:
Unfortunately, we are often using an `IWearRemapper` device in between the producer device, and the consumer device (in this case would be the `iwear_logger`), which merge data from multiple sources.

In this case, however, the state returned by `getStatus()` is the "worst status" that is found among all the associated sensors interface (see https://github.com/robotology/wearables/blob/master/devices/IWearRemapper/src/IWearRemapper.cpp#L802-L845). As a consequence, the specific status of the sensors interface is hidden. 

I think we can eventually discuss the current choice in the `IWearRemapper`, and change it.

Reply:
These lines were a copy-paste from [iWearWrapper](https://github.com/robotology/wearables/blob/c6a8478343881468b91112244bd25c6a803d456e/wrappers/IWear/src/IWearWrapper.cpp#L151-L155). Do you think this might not be needed here?

Reply:
> These lines were a copy-paste from [iWearWrapper](https://github.com/robotology/wearables/blob/c6a8478343881468b91112244bd25c6a803d456e/wrappers/IWear/src/IWearWrapper.cpp#L151-L155). Do you think this might not be needed here?

No, I think is fine as it is. I would rather change some logic on the Remapper side